src/Controls/src/Core/Application/Application.cs

```diff
@@ -85,13 +85,15 @@ public IAppLinks AppLinks
 		/// <include file="../../docs/Microsoft.Maui.Controls/Application.xml" path="//Member[@MemberName='MainPage']/Docs/*" />
 		public Page? MainPage
 		{
+			[Obsolete("This property has been deprecated. For single-window applications, use Windows[0].Page. For multi-window applications, identify and use the appropriate Window object to access the desired Page. Additionally, each element features a Window property, accessible when it's part of the current window.")]
 			get
 			{
 				if (Windows.Count == 0)
 					return _singleWindowMainPage;
 
 				return Windows[0].Page;
 			}
+			[Obsolete("This property is deprecated. Initialize your application by overriding Application.CreateWindow rather than setting MainPage. To modify the root page in an active application, use Windows[0].Page for applications with a single window. For applications with multiple windows, use Application.Windows to identify and update the root page on the correct window.  Additionally, each element features a Window property, accessible when it's part of the current window.")]
```

Comment:
Thinking out loud - this will cause a warning (or possibly error) in 100% of MAUI applications ever written, no? Is it that bad to keep this? Maybe an analyzer could be used to flag it and rewrite the code for those interested? (And ignored by those who don't.)

Reply:
Yea, I just worry about also dragging the past along with us and not overall widening the pit of success

From a library author perspective using these APIs is really bad. For example, none of this code from our toolkit will work for multi-window https://github.com/CommunityToolkit/Maui/blob/cc1fb686815587076bbad45a1903f3ec3039315d/src/CommunityToolkit.Maui/PopupService.cs#L17

We probably see an issue a month for android where users need to override `CreateWindow`

https://github.com/dotnet/maui/issues/23645
https://github.com/dotnet/maui/issues/23493

I also worry about new users using `MainPage` and ignoring the analyzer suggestions. 

The fix here is basically to do a solution search for 
MainPage and replace it with Windows[0].Page

I get it because how widespread it is that users will all see warnings, but, it just seems to put MAUI in general on a path to longer term refinement/success.

I could see us maybe Obsoleting the set and just using an analyzer on the `get` perhaps?